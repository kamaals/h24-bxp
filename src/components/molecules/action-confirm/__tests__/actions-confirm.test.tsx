import { describe, expect, it, beforeAll } from "vitest";
import { render, screen, cleanup, waitFor } from "@testing-library/react";
import ActionConfirm from "../action-confirm";
import userEvent from "@testing-library/user-event";

describe(ActionConfirm.name, async () => {
  beforeAll(() => {
    cleanup();
  });

  it("should match the snapshot", () => {
    const result = render(<ActionConfirm />);
    expect(result).toMatchSnapshot();
  });

  it("Should click trigger", async () => {
    const user = userEvent.setup();
    render(<ActionConfirm />);
    const btn = screen.getAllByTestId("action-confirm-trigger");
    await waitFor(async () => {
      await user.click(btn[0]);
      const okButton = screen.getByTestId("action-confirm-ok");
      expect(okButton).toBeInTheDocument();
      await user.click(okButton);
    });
  });

  it("Should click trigger cancel", async () => {
    const user = userEvent.setup();
    render(<ActionConfirm />);
    const btn = screen.getAllByTestId("action-confirm-trigger");
    await waitFor(async () => {
      await user.click(btn[0]);
      const okButton = screen.getByTestId("action-confirm-cancel");
      expect(okButton).toBeInTheDocument();
      await user.click(okButton);
    });
  });
});
