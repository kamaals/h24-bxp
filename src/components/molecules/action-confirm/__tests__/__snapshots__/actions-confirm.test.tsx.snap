// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ActionConfirm > should match the snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="w-full min-h-8 px-2 py-1.5 flex items-center overflow-hidden relative rounded-md"
      >
        <div
          aria-label="Delete"
          class="absolute z-1 flex items-center justify-start w-full text-sm gap-2"
          data-testid="action-confirm-trigger"
          style="opacity: 0; transform: scale(0);"
        >
          <svg
            aria-hidden="true"
            class="lucide lucide-trash2 lucide-trash-2 w-4 h-4 text-red-600"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M3 6h18"
            />
            <path
              d="M19 6v14c0 1-1 2-2 2H7c-1 0-2-1-2-2V6"
            />
            <path
              d="M8 6V4c0-1 1-2 2-2h4c1 0 2 1 2 2v2"
            />
            <line
              x1="10"
              x2="10"
              y1="11"
              y2="17"
            />
            <line
              x1="14"
              x2="14"
              y1="11"
              y2="17"
            />
          </svg>
          Delete
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="w-full min-h-8 px-2 py-1.5 flex items-center overflow-hidden relative rounded-md"
    >
      <div
        aria-label="Delete"
        class="absolute z-1 flex items-center justify-start w-full text-sm gap-2"
        data-testid="action-confirm-trigger"
        style="opacity: 0; transform: scale(0);"
      >
        <svg
          aria-hidden="true"
          class="lucide lucide-trash2 lucide-trash-2 w-4 h-4 text-red-600"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M3 6h18"
          />
          <path
            d="M19 6v14c0 1-1 2-2 2H7c-1 0-2-1-2-2V6"
          />
          <path
            d="M8 6V4c0-1 1-2 2-2h4c1 0 2 1 2 2v2"
          />
          <line
            x1="10"
            x2="10"
            y1="11"
            y2="17"
          />
          <line
            x1="14"
            x2="14"
            y1="11"
            y2="17"
          />
        </svg>
        Delete
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
